gap/SnAnUnknownDegree.gi

```diff
@@ -0,0 +1,312 @@
+# Input: Group G, upper error bound eps, upper degree bound N
+#
+# The following algorithm constructs a set of possible 3-cycles. It is based
+# on the simple observation that the product of two involutions t1, t2, which
+# only move one common point, squares to a 3-cycle.
+#
+# Returns a an iterator
+# TODO: take care of duplicate candidates?
+BindGlobal("ThreeCycleCandidatesIterator",
+    function(G, eps, N, groupIsOne, groupIsEq)
+    local
+        # involution
+        t,
+        # integers, controlling the number of iterations
+        M, B, T, C, logInt2N,
+        # integer, prime, loop variable
+        p,
+        # counters
+        nrInvolutions, nrTriedConjugates, nrThreeCycleCandidates,
+        # helper functions
+        tryThreeCycleCandidate, oneThreeCycleCandidate;
+    # Step 1: Initialization
+    # The current involution t_i
+    t := fail;
+
+    # Constants
+    # TODO: better iteration over primes
+    M := 1;
+    p := 3;
+    while p <= N do
+        M := M * p ^ LogInt(N, p);
+        p := NextPrimeInt(p);
+    od;
```

Comment:
How about adding and using this helper function (based on a code snippet by @rbehrends )
```
AllPrimesUpTo := function(n)
    local i, j, sieve, result;
    if n <= 1000 then
      return Primes{[1..PositionSorted(Primes, n+1)-1]};
    fi;
    sieve := BlistList([1..n], [1..n]);
    sieve[1] := false;
    for i in [2..Int(n/2)] do
        sieve[i*2] := false;
    od;
    i := 3;
    while i * i <= n do
        if sieve[i] then
            j := 3*i;
            while j <= n do
                sieve[j] := false;
                j := j + 2*i;
            od;
        fi;
        i := i + 2;
    od;
    return ListBlist([1..n], sieve);
end;
```

Reply:
Done.

Reply:
done